src/Analyzers/MSTest.Analyzers/ClassInitializeShouldBeValidAnalyzer.cs

```diff
@@ -75,6 +79,36 @@ public override void Initialize(AnalysisContext context)
             return;
         }
 
+        if (namedTypeSymbol.IsGenericType)
+        {
+            bool isInheritanceModeSet = false;
+            foreach (AttributeData attr in methodSymbol.GetAttributes())
+            {
+                ImmutableArray<TypedConstant> constructorArguments = attr.ConstructorArguments;
+
+                foreach (var constructorArgument in constructorArguments)
+                {
+                    // Null is considered as default for non-nullable types.
+                    if (constructorArgument.IsNull)
+                    {
+                        continue;
+                    }
+
+                    // We need to check that the inheritanceBehavior is not set to none and it's value inside the enum is zero
+                    if (SymbolEqualityComparer.Default.Equals(constructorArgument.Type, inheritanceBehaviorSymbol)
+                        && constructorArgument.Value?.ToString() != "0")
+                    {
+                        isInheritanceModeSet = true;
```

Comment:
can we "break" as soon as we find the value? Or it's possible that another one will set it false?